build: fix verify-archive

benesch opened the pull request:
verify-archive has been broken since August, but is quite useful
ensuring we don't accidentally introduce new build-time dependencies.

Fix it up by removing the Git dependency from compiling protobufs.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/19506)
<!-- Reviewable:end -->

benesch commented:
TeamCity's verify archive build: https://teamcity.cockroachdb.com/viewType.html?buildTypeId=Cockroach_UnitTests_VerifyArchive&branch_Cockroach_UnitTests=19506&tab=buildTypeStatusDiv

lego commented on build/protobuf.mk:
```diff
@@ -30,7 +30,7 @@ PROTOC_PLUGIN   := $(LOCAL_BIN)/protoc-gen-$(PLUGIN_SUFFIX)
 
 COREOS_PATH := $(GITHUB_ROOT)/coreos
-COREOS_RAFT_PROTOS := $(addprefix $(COREOS_PATH)/etcd/raft/, $(sort $(shell git -C $(COREOS_PATH)/etcd/raft ls-files --exclude-standard --cached --others -- '*.proto')))
+COREOS_RAFT_PROTOS := $(sort $(shell find $(COREOS_PATH)/etcd/raft -type f -name '*.proto'))
```
OOC, any reason for the change to find? Performance, consistency, or clarity?

benesch replied:
The source tarballs strip the Git checkout to reduce the size of the tarball. (The `.git` directory on my machine is 1.6GB, which is _crazy_.) So `git ls-files` doesn't work, but `find` does. This `git ls-files`wasn't an issue before because `make build` didn't call `make protobuf`, but the removal of embedded.go means every `make build` invocation checks to make sure the protobufs are up-to-date.

lego commented on build/verify-archive.sh:
```diff
@@ -2,8 +2,9 @@
 # environment with a Go version capable of building CockroachDB. Source tarballs
-# are expected to build and install a functional cockroach binary into the PATH,
-# even when the tarball is extracted outside of GOPATH.
+# are expected to build, build after cleaning, and install a functional
```
Is the purpose of the clean-then-build to ensure we don't clean up the new precompiled dependencies  (`embedded.go`, etc.)?

Is there any reason for doing the first build?

benesch replied:
Heh, good point. Changed to clean then build.

lgo reviewed (approved):
LGTM, but it looks like verify-archive is still failing.

https://teamcity.cockroachdb.com/viewLog.html?buildId=388452&buildTypeId=Cockroach_UnitTests_VerifyArchive&tab=buildLog&_focus=1442
```
./cockroach/build/node-run.sh: line 42: yarn: command not found
```

benesch committed:
build: fix verify-archive

verify-archive has been broken since August and accordingly disabled on
TeamCity. It's quite useful ensuring we don't accidentally introduce new
build-time dependencies, though, so fix it up by:

  * Installing the new compile-time dependency, libtinfo-dev, in the
    verification Docker container.

  * Removing the Git dependency from compiling protobufs by using `find`
    instead of `git ls-files` in the Makefile.

  * Removing the Node and Yarn dependency when bindata.go is up-to-date
    by marking intermediate files as such in the Makefile.
